config/translate.go

```diff
@@ -165,25 +180,53 @@ func TranslateFromV1(old v1.Config) types.Config {
 	return config
 }
 
+// golang--
+func translateV1MkfsOptionsToV2_1OptionSlice(opts v1.MkfsOptions) []types.Option {
+	newOpts := make([]types.Option, len(opts))
+	for i, o := range opts {
+		newOpts[i] = types.Option(o)
+	}
+	return newOpts
+}
+
+// golang--
+func translateStringSliceToV2_1SSHAuthorizedKeySlice(keys []string) []types.SSHAuthorizedKey {
+	newKeys := make([]types.SSHAuthorizedKey, len(keys))
+	for i, k := range keys {
+		newKeys[i] = types.SSHAuthorizedKey(k)
+	}
+	return newKeys
+}
+
+// golang--
+func translateStringSliceToV2_1UsercreateGroupSlice(groups []string) []types.UsercreateGroup {
+	var newGroups []types.UsercreateGroup
```

Comment:
An alternate option would be to preallocate, but at the beginning have `if groups == nil return nil` (or len == 0 if it turns out the test wants [] -> nil, not just nil -> nil).

Thanks for explaining, this comment's just a nit.